R/course.R

```diff
@@ -259,6 +264,81 @@ tidy_download <- function(url, destdir = getwd()) {
   invisible(full_path)
 }
 
+download_url <- function(url,
+                         destfile,
+                         handle = curl::new_handle(),
+                         n_tries = 3,
+                         retry_connecttimeout = 40L) {
```

Comment:
Our first try is always with the curl default. The curl package defaults to 10s, which was clearly too short at rstudio::conf. 

All retries are with `retry_connecttimeout`. I chose 40 seconds because we (and others on the internet) seem to talk about values in the range of 30 to 60 second.

There are other knobs that affect timeout (see #988 for research), but the curl package doesn't set its own default for anything but `connecttimeout` and I believe it's the one that caused us grief. I'd only consider messing with the others if we have evidence we need to.